### PR TITLE
Ft add cuboid

### PR DESCRIPTION
**Pull request by GijsGroote**
2 test in test_boxObstacle.py are failing. This is because the BoxObstacle does not check if the shape of color and orientation is correct.

What is a nice place to do input validation for color and orientation?

### CODE REVIEW

File: MotionPlanningEnv/sphereObstacle.py [hunk 1]
```diff
@@ -87,14 +90,14 @@ def limitHigh(self):
...
 
-    def velocity(self, **kwargs):
-        return np.zeros(self.dim())
+    def velocity(self):
+        raise NotImplementedError
 
-    def acceleration(self, **kwargs):
-        return np.zeros(self.dim())
+    def acceleration(self):
+        raise NotImplementedError
```
Comment (maxspahn) [inline MotionPlanningEnv/sphereObstacle.py:100]: Why is this change needed?

File: MotionPlanningEnv/sphereObstacle.py [hunk 2]
```diff
@@ -138,18 +141,17 @@ def add2Bullet(self, pybullet):
...
             pybullet.GEOM_MESH,
-            fileName='sphere_smooth.obj',
-            rgbaColor=[1.0, 0.0, 0.0, 1.0],
-            specularColor=[1.0, 0.5, 0.5],
+            fileName='sphere.obj',
+            rgbaColor=self.color(),
             meshScale=[self.radius(), self.radius(), self.radius()]
         )
-        pybullet.createMultiBody(mass,
+        pybullet.createMultiBody(self.mass(),
```
Comment (maxspahn) [inline MotionPlanningEnv/sphereObstacle.py:154]: I will have to check if it breaks the movable feature. Because the default value is changed to `1` and `-1` would be result in a fixed obstacle.

File: MotionPlanningEnv/cilinder.obj [hunk 3]
```diff
@@ -0,0 +1,268 @@
+# Blender v3.2.2 OBJ File: 'untitled.blend'
+# www.blender.org
+mtllib cilinder.mtl
```
Comment (maxspahn) [inline MotionPlanningEnv/cilinder.obj:3]: Could you change this to be cylinder instead of the dutch spelling?

**Review by maxspahn — CHANGES_REQUESTED** [review]
Nice work. However, I would prefere to a have PR that only changes one thing instead of multiple things. Here, sphereObstacle is changed while the name suggest that only a box/cuboid is added.

Could you stick to english spelling.
Can you verify whether the movable feature still works, especially the case when it should not be movable?

**Comment by GijsGroote** [comment]
PR should do one thing, I agree. I will take that into account in future pull requests.

I will make some changes this evening, hopefully, there is still some time to check them. 

Anyhow, enjoy the Holliday.

File: MotionPlanningEnv/sphereObstacle.py (hunk 1 above)
Comment (GijsGroote) [reply MotionPlanningEnv/sphereObstacle.py:100]: Obstacles could have an initial velocity or acceleration, however, this function now indicates it does not have initial velocity or acceleration. 

I think velocity and acceleration could entirely be skipped unless in the future someone would like initial vel/acc.

Do you want to keep track of vel/acc while the simulation is running? Or mostly for initial vel/acc?

File: MotionPlanningEnv/sphereObstacle.py (hunk 2 above)
Comment (GijsGroote) [reply MotionPlanningEnv/sphereObstacle.py:154]: A negative mass is not allowed any more. 
side note: any mass which is 0 or smaller than 0 makes the obstacle fixed

File: MotionPlanningEnv/cilinder.obj (hunk 3 above)
Comment (GijsGroote) [reply MotionPlanningEnv/cilinder.obj]: The name is updated to cylinder

**Comment by GijsGroote** [comment]
THIS PULL REQUEST WILL BE SPLIT UP IN MULTIPLE PR's !

Branch: ft-adding-obstacles will be used for making changes -> create PR to main -> create changes -> create PR to main -> etc.
This explains the (not so great) name ft-adding-obstacles, because in some PR it does not add obstacles.
image
